### DISCUSSION TITLE
dots-hyprland on Gentoo

### DISCUSSION THREAD

**Discussion by Demiurg39** [end-4/dots-hyprland#997] [Extra Distros]
Hey guys make set for deps. Most of packages available on gentoo and guru repos. Set also contain font and gtk theme but you need manualy add it to your local repo(or just use manual_install_script). Also  pythonmaterialcolor was a pain to just ebuild it but seems theres some deps so if you want to install this try it by adding gentooplus repo. I wasnt found ebuild for MicroTex so i just install texlive. Also i still not tested if dotfiles work, so it might not working. If i make this thing work, will update post

1. Enable guru repo
>`sudo eselect repository enable guru`

2. Then create sets directory
>`sudo mkdir /etc/portage/sets`

3. Place illogical_impulse.txt at previously created directory
>`sudo mv ~/Downloads/illogical-impulse.txt /etc/portage/sets/illogical-impulse`

4. Optional: uncoment fonts and themes (if you choose to install them via local repo)

5. Then we can install deps
> `sudo emerge -a @illogical-impulse`

6. After that you could exec manual_install_helper

### Ebuilds (font, gtk, etc.)
[illogical-impulse.txt](https://github.com/user-attachments/files/18183220/illogical-impulse.txt)
[adw-gtk-5.6.txt](https://github.com/user-attachments/files/18116647/adw-gtk-5.6.txt)
[anyrun-241108.txt](https://github.com/user-attachments/files/18116649/anyrun-241108.txt)
[blueberry-1.4.8.txt](https://github.com/user-attachments/files/18116650/blueberry-1.4.8.txt)
[dart-sass-1.81.0.txt](https://github.com/user-attachments/files/18116651/dart-sass-1.81.0.txt)
[gabarito-1.000.txt](https://github.com/user-attachments/files/18116652/gabarito-1.000.txt)
[jetbrains-mono-nerd-3.0.2.txt](https://github.com/user-attachments/files/18116653/jetbrains-mono-nerd-3.0.2.txt)
[space-mono-nerd-3.3.0.txt](https://github.com/user-attachments/files/18116655/space-mono-nerd-3.3.0.txt)
[material-design-icons-4.0.0.txt](https://github.com/user-attachments/files/18183221/material-design-icons-4.0.0.txt)
[oneui4-icons-9999.txt](https://github.com/user-attachments/files/18183222/oneui4-icons-9999.txt)
[readexpro-9999.txt](https://github.com/user-attachments/files/18183223/readexpro-9999.txt)
[rubik-9999.txt](https://github.com/user-attachments/files/18183224/rubik-9999.txt)

**Comment by Poomrapee-chsk** [comment]
Tried this and it works. Thanks a lot!
I would recommend using hyprland-overlay for materialyoucolor. 
Also, install material-design-symbols font for the symbols in the AGS bar.

**Comment by clsty** [comment]
> Also pythonmaterialcolor was a pain to just ebuild it

I just noticed this... Well currently we're using `uv` to install python packages into virtual env, which path is defined as env var `$ILLOGICAL_IMPULSE_VIRTUAL_ENV` and by default it is `~/.local/state/ags/.venv`.

Specifically in `manual-install-helper.sh` there're two functions dealing with it: `install-uv` and `install-python-packages`.

**Comment by dawidrogowicz** [comment]
I will also share set I ended up with after doing a minimal setup.

Goal was to have just the main ags bar with theme generation working but I also decided to keep screenshot/recording utils.

Most packages are available in guru/hyprland-overlay + anyrun/dart-sass from xoores
[material-design-symbols](https://github.com/google/material-design-icons/tree/master/font) manually downloaded and placed in `.local/share/fonts/`
<details>
<summary>/etc/portage/sets/dots-hyprland-minimal</summary>

```
# ags and its deps
gui-apps/ags
x11-themes/adw-gtk3
x11-misc/xdg-user-dirs # could be removed and just hardcode paths in ags
gui-apps/yad
x11-themes/kvantum #qt theming
app-misc/ddcutil
dev-util/dart-sass
dev-libs/libportal
dev-libs/libdbusmenu

# main apps
gui-apps/anyrun
net-misc/networkmanager # Potentially removable if using ethernet

# wayland utils (also used by ags)
net-wireless/gnome-bluetooth
app-misc/brightnessctl
gui-apps/wl-clipboard # clipboard
gui-apps/swww # wallpaper switcher

# audio control
media-sound/playerctl
media-sound/pavucontrol
media-sound/easyeffects

# waifu sidebar
x11-libs/gdk-pixbuf

#screenshots/recording
gui-apps/slurp
gui-apps/grim
gui-apps/swappy
gui-apps/wf-recorder

# fonts and cursors (could be downloaded from scripts)
x11-themes/oneui4-icons
x11-themes/bibata-xcursors
dev-libs/tinyxml2 # for microtex compilation
dev-cpp/gtksourceviewmm # for microtex compilation
```
</details>

**Comment by clsty** [comment]
Update:

- I have reworked install script to handle different distros support.
- Thanks to @jwihardi, `dist-gentoo` has been added in #2116.

As the result, **now you can use `./setup install` for Gentoo.**

**Comment by josser** [comment]
@Demiurg39 Maybe it’s not a good idea to include coreutils in the set. Every system already has it installed anyway, but if someone tries to remove @set, it’ll also remove coreutils and break the whole system.

**Reply by clsty** [reply]
Really? This seems a lot different than normal "meta package", like in Arch Linux. If we uninstalled all `illogical-impulse-*` Arch packages, the dependency which it contains will not get removed as long as it's also required by other packages or itself has been installed as explicitly.

**Reply by josser** [reply]
Never used sets before yesterday, so I'm not sure is it my fault or correct behavior ) I run into this myself :D 
Maybe coreutils is not dependent on anything on my system (quite fresh install). I believe it should be part of @system but maybe not. And that's why portage uninstalled it.
